Feature #155 Creating the sign up page in frontend

CenekSanzak opened the pull request:
Issue #155 I created the signup page for the frontend project. It has input validation with joi. I created a feedback mechanism using Alert components of material ui. It doesn't work with the backend yet, just frontend. I took a screenshot for demonstration:
![image](https://user-images.githubusercontent.com/36790615/140993566-781c3875-5548-4ef8-99a4-46245e35d169.png)

berkaydoner reviewed (changes requested):
Thank you! I think overall design is very good; however, you can enhance some parts:
1-) In the password check part, password is checked with rule 
```Joi.string().pattern(new RegExp('^[a-zA-Z0-9]{3,30}$'))```
I think this forces passwords to be at least 3 characters long while it should be at least 8 characters long, based on the requirement 2.5.1. Other parts of this requirement are satisfied with this pattern, so no need to add anything. 
2-) I think you can discard the "Built by reBOUNd team.". We can design a more detailed footer component in the future, we can also put this info there.
3-) Now that we have a logo, you can put it instead of lock icon.
4-) I think it might look better if you can make the "Already have an account?" part regular text, not a link. You can just leave it be if you want though.

CenekSanzak commented:
I fixed the minimum length for password. I removed the footer. "Already have an account?" text is now looks like a normal text but it navigates to signup page. This way it looks better. I also added our new logo.

berkaydoner reviewed (approved):
I think these changes are enough and you can merge the changes. Connection with backend can be discussed in a separate branch&pull request.
